appHooks/appHook.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * appHooks/appHook.js
+ * Construction of all appHooks
+ *
+ * @exports {Object} default - an appHook
+ */
+
+import _ from 'lodash'
+import Promise from 'bluebird'
+
+import { Helpers, Log, Request } from '~/app'
+
+let fs = Promise.promisifyAll(require('fs'))
+
+class AppHook {
+  constructor (data, obj) {
+    this.data = data
+
+    this.name = obj.name || 'appHook'
+    this.path = obj.path || `${__dirname}/${this.name}`
+    this.mark = obj.mark || 'issue.md'
+    this.post = obj.post || false
+
+    this.errors = []
+    this.warnings = []
+    this.information = {}
+  }
+
+  test (data) {
+    Log.warn(`${this.name} does not have any test`)
+    return
+  }
+
+  error (msg) {
+    this.errors.push(msg)
+  }
+
+  warn (msg) {
+    this.warnings.push(msg)
+  }
+
+  update (obj) {
+    this.information = _.extend(this.information, obj)
```

Comment:
in ES6, you can use `Object.assign()`. would also mean you can ditch the _ require in this file :)